src/main/kotlin/no/nav/tilleggsstonader/sak/statistikk/behandling/BehandlingsstatistikkService.kt

```diff
@@ -0,0 +1,207 @@
+                null
+            },
+            vedtakTid = if (Hendelse.VEDTATT == hendelse) {
+                hendelseTidspunkt.atZone(europeOslo)
+            } else {
+                null
+            },
+            ferdigBehandletTid = if (Hendelse.FERDIG == hendelse) {
+                hendelseTidspunkt.atZone(europeOslo)
+            } else {
+                null
+            },
+            totrinnsbehandling = totrinnskontrollErGodkjent(behandlingId),
+            sakUtland = mapTilStreng(saksbehandling.kategori),
+            relatertBehandlingId = relatertEksternBehandlingId,
+            versjon = Applikasjonsversjon.versjon,
+            vilkårsprøving = emptyList(), // TODO: Implementer dette i samarbeid med Team SAK. Ikke kritisk å ha med i starten.
+            revurderingÅrsak = null, // TODO aktiver når revurdering er implementert
+            revurderingOpplysningskilde = null, // TODO aktiver når revurdering er implementert
+        )
+    }
+
+    private fun settBeslutterId(
+        hendelse: Hendelse,
+        behandlingId: UUID,
+    ): String? {
+        val beslutterId = if (hendelse.erBesluttetEllerFerdig()) {
+            totrinnskontrollService.hentBeslutter(behandlingId)
+        } else {
+            null
+        }
+        return beslutterId
+    }
+
+    private fun totrinnskontrollErGodkjent(behandlingId: UUID): Boolean {
+        val totrinnskontrollstatus = totrinnskontrollService.hentTotrinnskontrollStatus(behandlingId).status
+        return when (totrinnskontrollstatus) {
+            TotrinnkontrollStatus.GODKJENT -> true
+            else -> false
+        }
+    }
+
+    private fun finnSisteOppgaveForBehandlingen(behandlingId: UUID, oppgaveId: Long?): Oppgave? {
+        val gsakOppgaveId = oppgaveId ?: oppgaveService.finnSisteOppgaveForBehandling(behandlingId)?.gsakOppgaveId
+
+        return gsakOppgaveId?.let { oppgaveService.hentOppgave(it) }
+    }
+
+    private fun Hendelse.erBesluttetEllerFerdig() =
+        this.name == Hendelse.BESLUTTET.name || this.name == Hendelse.FERDIG.name
+
+    private fun finnSaksbehandler(
+        hendelse: Hendelse,
+        totrinnskontrollService: TotrinnskontrollService,
+        gjeldendeSaksbehandler: String?,
+        behandlingId: UUID,
+    ): String {
+        return when (hendelse) {
+            Hendelse.MOTTATT, Hendelse.PÅBEGYNT, Hendelse.VENTER, Hendelse.HENLAGT ->
+                gjeldendeSaksbehandler ?: error("Mangler saksbehandler for hendelse")
+
+            Hendelse.VEDTATT, Hendelse.BESLUTTET, Hendelse.FERDIG ->
+                totrinnskontrollService.hentSaksbehandlerSomSendteTilBeslutter(behandlingId)
+        }
+    }
+
+    private fun finnHenvendelsestidspunkt(saksbehandling: Saksbehandling): LocalDateTime {
+        return when (saksbehandling.type) {
+            BehandlingType.FØRSTEGANGSBEHANDLING -> saksbehandling.opprettetTid
+            BehandlingType.REVURDERING -> saksbehandling.opprettetTid
+        }
+    }
```

Comment:
Burde denne egentlig se på søknadstidspunktet? For automatisk journalførte behandlinger vil det omtrent være det samme. Men for ting som må manuelt journalfører kan det være mange ukers forskjell. Ser fra kontakten at det "... Det er ønskelig å måle brukers opplevde ventetid."

Reply:
Burde kanskje kunne bruke `saksbehandling.kravmottatt` men ser ikke ut som den alltid er populert 😞

Reply:
kravMottatt var ein av dei eg kikka på der, men, mulig at me kan sette kravMottat som primær, og så heller ha fallback til opprettetTid om kravMottatt mangler.

Reply:
Vi byttet til mottattTidspunkt for søknaden. Ser det OK ut @eirarset ?

Reply:
Supert 💯